### PR TITLE
Improve Kafka Listener documentation

### PR DESCRIPTION
**Pull request by matthew-d-jones**
**Description of work.**

Added documentation on configuration of topics for Kafka listener algorithms following questions from ANSTO.
FYI @geishm-ansto 

I also removed the detector-spectrum map topic name which is not actually being used any more but was still being passed around to the Kafka listener algorithms.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: docs/source/concepts/KafkaLiveStreams.rst [hunk 1]
```diff
@@ -43,5 +43,58 @@ This following python script shows how this listener is used:
       Listener='KafkaHistoListener', Address='kafkabroker:address', AccumulationMethod='Replace',
       RunTransitionBehavior='Stop', OutputWorkspace='testout')
 
+Topic Configuration
+###################
+
+The topics that the listener will subscribe to are defined for a particular instrument in Mantid's "Facilities.xml" file. Here is the entry for the V20 beamline for example:
```
Comment (OwenArnold) [inline docs/source/concepts/KafkaLiveStreams.rst:49]: Could you point to this in the source code maybe. The link is probably not fragile given that the location has remained unchanged in a long long time.

File: docs/source/concepts/KafkaLiveStreams.rst [hunk 2]
```diff
@@ -43,5 +43,58 @@ This following python script shows how this listener is used:
...
+
+.. code-block:: xml
+
+    <instrument name="V20" shortname="V20" >
+      <zeropadding size="8" />
```
Comment (OwenArnold) [inline docs/source/concepts/KafkaLiveStreams.rst:54]: Can we take things out that are irrelevant. I think zero-padding is not needed?

File: docs/source/concepts/KafkaLiveStreams.rst [hunk 3]
```diff
@@ -43,5 +43,58 @@ This following python script shows how this listener is used:
...
+   * - event
+     - ev42
+     - Yes (but topic can be empty)
+     - Detection event data
+   * - monitor
+     - ev42
+     - Yes (but topic can be empty)
+     - Detection event data from monitors (single pixel detectors). This just allows using a separate topic for these data, alternatively they can be published to the "event" topic with other data from other detectors.
+
+Note, there must be a run start message (schema pl72) available in the "run" topic for the listener to start. If the "nexus_structure" field of this message contains geometry information in NeXus format (NXoff_geometry or NXcylindrical_geometry) then Mantid will parse this to get the instrument geometry and expected detector ids etc. Otherwise it uses the "instrument_name" to look up a Mantid Instrument Definition File (IDF) for the instrument.
```
Comment (OwenArnold) [inline docs/source/concepts/KafkaLiveStreams.rst:95]: I don't feel that there is enough information between this and https://github.com/ess-dmsc/streaming-data-types about what the schemas relate to, what the short codes mean etc.

Comment (OwenArnold) [inline docs/source/concepts/KafkaLiveStreams.rst:95]: Minor point: Regarding the "lookup" for instrument information. These are basically the same rules that `LoadInstrument` uses. You might wan to cross-reference.

File: docs/source/concepts/KafkaLiveStreams.rst [hunk 4]
```diff
@@ -63,6 +62,13 @@ The topics that the listener will subscribe to are defined for a particular inst
       </livedata>
     </instrument>
 
+Data in the Kafka topics is serialised using Google FlatBuffers, according to schema which can be found in the `ESS Streaming-Data-Types repository <https://github.com/ess-dmsc/streaming-data-types>`.
+The FlatBuffer Compiler tool generates C++ code for each schema file to provide an implementation for serialising and deserialising data to the format it is communicated in, over the network, through Kafka.
+These generated C++ files are included in the Mantid source.
+Each schema is identified by a four character string, for example "ev42" which identifies the schema defining the serialised data format for detection event data.
+The schema identifiers are defined in their corresponding schema file, and are included in the schema file name and the generated C++ code filenames.
+Particular serialised data are expected to be found in different topics on Kafka. The schema identifier, or identifiers, for data in each topic "type" are documented in the table below.
+
```
Comment (OwenArnold) [inline docs/source/concepts/KafkaLiveStreams.rst:71]: 👍

**Review by OwenArnold — APPROVED** [review]
I don't want to hold this up any longer. The two failing tests look unrelated, ive re-run those builds for good measure.

**Comment by martyngigg** [comment]
Thanks for updating this. It makes things much clearer and it's good to remove the unused code 👍🏻

**Pull request merged**